### PR TITLE
Update dojo/loader to use TypeScript 1.8.2

### PR DESCRIPTION
**Pull request by Tomdye**
Fixes: https://github.com/dojo/loader/issues/45

- Had to pass the `noImplicitUseStrict` argument via the `grunt-ts` `additionalFlags` attribute.
- Changed `var` for `let`
- Used the latest `tslint` and `package.json` configs from the template

### CODE REVIEW

File: src/loader.ts [hunk 1]
```diff
@@ -385,6 +389,40 @@ interface ModuleDefinitionArguments extends Array<any> {
 		pendingCacheInsert = {};
 	}
 
+	function loadNodeModule(moduleId: string, parent?: Module): any {
```
Comment (kitsonk) [inline src/loader.ts]: Why did we move this?  In the future, when we have a builder that can optimise based on has flags, this won't now be elided.

File: .gitignore [hunk 2]
```diff
@@ -14,3 +14,4 @@ tests/typings/dist/
 html-report
 coverage*.json
 lcov.info
+.vscode
```
Comment (kitsonk) [inline .gitignore]: Generally, things like this we should add it to our global `.gitignore` (anything specific to the editors, or local filesystem, like `.DS_Store`).  I may have forgotten to do this in the past myself.

To set it up, check out [Create Global Git Ignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore)

Comment (Tomdye) [inline .gitignore]: ok no worries, can put that in

File: src/loader.ts (hunk 1 above)
Comment (Tomdye) [inline src/loader.ts]: it was out of scope when the new global stuff was introduced, couldn't fully determine why, but I believe it was garbage collected maybe?

Comment (kitsonk) [inline src/loader.ts]: We should still find a way to `has()` feature this.  One simple solution is to create a `let loadNodeModule: (moduleId: string, parent?: Module) => any = () => { };` and then assign it where it originally was.  If we have a `noop` handle somewhere, that would be better...

Comment (Tomdye) [inline src/loader.ts]: no worries, will do the same for https://github.com/dojo/loader/pull/46/files#diff-1d670561cb1689a37382f052a97a2e97R908

**Pull request closed without merging**